lib/range.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * ...something resembling a binary search, to find the lowest line within the range.
+ * And then you could break as soon as the line is longer than the range...
+ */
+module.exports.sliceRange = (lines, startCol, endCol) => {
+  let lower = 0
+  let upper = lines.length - 1
+
+  let s
+  while (lower <= upper) {
```

Comment:
thank you, I actually spent some time asking myself: is it better to do 2 binary searches or not? At the end I preferred to opt for only 1 binary search and then linear scan because it's unlikely that in a large file a range would be too big anyway so I guess that the linear search should perform better